hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -601,37 +601,107 @@ def hpat_pandas_series_isin_impl(self, values):
+            Given to_append: {}'.format(_func_name, to_append))
+
+    # currently we will always raise this in the end, i.e. if no impl was found
+    # TODO: find a way to stop compilation early and not proceed with unliteral step
+    if not (isinstance(ignore_index, types.Literal) and isinstance(ignore_index, types.Boolean)
+            or isinstance(ignore_index, types.Omitted)
+            or ignore_index is False):
+        raise TypingError(
+            '{} The ignore_index must be a literal Boolean constant. Given: {}'.format(_func_name, ignore_index))
+
+    if not (verify_integrity is False or isinstance(verify_integrity, types.Omitted)):
         raise TypingError(
-            '{} The object must be a pandas.series. Given self: {}, to_append: {}'.format(_func_name, self, to_append))
+            '{} Unsupported parameters. Given verify_integrity: {}'.format(_func_name, verify_integrity))
+
+    # ignore_index value has to be known at compile time to select between implementations with different signatures
+    if ((isinstance(ignore_index, types.Literal) and ignore_index.literal_value is True)
+            or (isinstance(ignore_index, bool) and ignore_index is True)):
+        # implementations that ignore series index
+        if isinstance(to_append, SeriesType):
+            def hpat_pandas_series_append_single_impl(self, to_append, ignore_index=False, verify_integrity=False):
+
+                new_data = hpat.hiframes.api._append(self._data, to_append._data)
+                new_index = numpy.arange(len(self._data) + len(to_append._data))
+                return pandas.Series(new_data, new_index)
+
+            return hpat_pandas_series_append_single_impl
+
+        elif isinstance(to_append, (types.UniTuple, types.List)):
+            def hpat_pandas_series_append_list_impl(self, to_append, ignore_index=False, verify_integrity=False):
+
+                arrays_to_append = [series._data for series in to_append]
+                sum_of_sizes = numpy.array([len(arr) for arr in arrays_to_append]).sum()
+                new_data = hpat.hiframes.api._append(self._data, arrays_to_append)
+                new_index = numpy.arange(len(self._data) + sum_of_sizes)
+                return pandas.Series(new_data, new_index)
+
+            return hpat_pandas_series_append_list_impl
+
+    elif ((isinstance(ignore_index, types.Literal) and ignore_index.literal_value is False)
+            or (isinstance(ignore_index, bool) and ignore_index is False)
+            or isinstance(ignore_index, types.Omitted)):
+        # implementations that handle series index (ignore_index is False)
+        if isinstance(to_append, SeriesType):
+            def hpat_pandas_series_append_single_impl(self, to_append, ignore_index=False, verify_integrity=False):
+
+                new_data = hpat.hiframes.api._append(self._data, to_append._data)
+                new_index = hpat.hiframes.api._append(self.index, to_append.index)
```

Comment:
Does it work with `self.index`? Shouldn't we use `self._index`?

Reply:
@densmirn Yes, self.index returns np.arange index if self._index is None. 
Thanks for the comment, I'll update my tests to check that series without index are handled too.

Reply:
@densmirn Actually, it turned out tests with suffix index_default (e.g. test_series_append_single_index_default) check exactly that self.index is created, while self._index is None (as we unbox default series index into None).